src/backend/distributed/commands/utility_hook.c

```diff
@@ -292,6 +294,21 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 		PG_RE_THROW();
 	}
 	PG_END_TRY();
+
+	if (AddAllLocalTablesToMetadata && ShouldConvertTable && !IsBinaryUpgrade &&
```

Comment:
could make sense to move this into the `if (UtilityHookLevel == 1)` block above, below the `if (ShouldCheckUndistributeCitusLocalTables())` block.